sql-plugin/src/main/scala/com/nvidia/spark/rapids/MetaUtils.scala

```diff
@@ -45,23 +45,81 @@ object MetaUtils {
       buffer)
   }
 
+  /**
+   * Build a TableMeta message
+   * @param tableId the ID to use for this table
+   * @param columns the columns in the table
+   * @param numRows the number of rows in the table
+   * @param buffer the contiguous buffer backing the columns in the table
+   * @return heap-based flatbuffer message
+   */
   def buildTableMeta(
       tableId: Int,
       columns: Seq[ColumnVector],
       numRows: Long,
       buffer: DeviceMemoryBuffer): TableMeta = {
     val fbb = new FlatBufferBuilder(1024)
-    val baseAddress = buffer.getAddress
     val bufferSize = buffer.getLength
-    val bufferMetaOffset = BufferMeta.createBufferMeta(
+    BufferMeta.startBufferMeta(fbb)
+    BufferMeta.addId(fbb, tableId)
+    BufferMeta.addSize(fbb, bufferSize)
+    BufferMeta.addUncompressedSize(fbb, 0)
```

Comment:
Just curious why the uncompressed size is 0. There are no docs anywhere that explain that uncompressed size is optional and in what ways it is optional.

Reply:
See [here](https://github.com/NVIDIA/spark-rapids/blob/10371d427cb2395e92f095a8711705c6c80d642d/sql-plugin/src/main/format/ShuffleCommon.fbs#L52) and [here](https://github.com/NVIDIA/spark-rapids/blob/10371d427cb2395e92f095a8711705c6c80d642d/sql-plugin/src/main/java/com/nvidia/spark/rapids/format/BufferMeta.java#L28).

Originally I was planning on using an uncompressedSize of 0 to indicate the data was not compressed with a codec, but I changed it to checking `codecBufferDescrsLength > 0` instead.  If we want this to match the size when the buffer is uncompressed that's an easy change to make.

Reply:
Sorry I missed those. It should be fine.

Reply:
`size` is always the size of the buffer, compressed or not.  The shuffle transport will only ever need to look at that value, since it is never going to uncompress the data directly rather just transport it as-is.

`uncompressedSize` is only useful for those who have already detected the buffer is compressed and are interested in decoding it.  If code doesn't want to know or care if the buffer is compressed, `size` is all they should ever need.

Reply:
In the case of inflight limit, if compression is really great, the transport may want to use the uncompressed size to throttle, rather than the compressed size. Otherwise it makes the OOM cases harder to configure for (how do I tell the user to pick an in-flight size that fits all their data/jobs?)